Mathlib/GroupTheory/FreeGroup/Orbit.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2025 Christian Krause. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Krause
+-/
+import Mathlib.GroupTheory.FreeGroup.Reduce
+import Mathlib.GroupTheory.GroupAction.Defs
+
+/-!
+For any `w : α × bool`, `FreeGroup.startsWith w` is the set of all elemenents of `FreeGroup α` that
+start with `w`.
+
+The main theorem `Orbit.duplicate` proves that applying `w⁻¹` to the orbit of `x` under the action
+of `FreeGroup.startsWith w` yields the orbit of `x` under the action of `FreeGroup.startsWith v`
+for every `v ≠ w⁻¹` (and the point `x`).
+-/
+
+variable {α X : Type*} [DecidableEq α]
+
+namespace FreeGroup
+
+/--
+All elements of the free Group that start with a certain letter.
+-/
+def startsWith (w : α × Bool) := {g : FreeGroup α | (FreeGroup.toWord g)[0]? = some w}
+
+/--
+The neutral element is not contained in one of the startsWith sets.
+-/
+theorem startsWith.ne_one {w : α × Bool} (g : FreeGroup α) (h : g ∈ FreeGroup.startsWith w) :
+    g ≠ 1 := fun h1 ↦ by simp [h1, startsWith, FreeGroup.toWord_one] at h
+
+theorem not_startsWith_of_ne {w w' : α × Bool} (hw : w ≠ w')
+    (g : FreeGroup α) (h : g ∈ FreeGroup.startsWith w) : ¬ g ∈ FreeGroup.startsWith w' := by
+  grind [startsWith]
```

Comment:
Hiding in here there seems to be a statement about injectivity of `startsWith`: would formulating such a statement simplify the arguments?

Reply:
I like the version proving disjointness!  Also, technically this leaves the possibility that everything is mapped to the empty set, failing injectivity, but if this is what you need, then this is the "correct" lemma!